clinica/utils/inputs.py

```diff
@@ -329,7 +334,20 @@ def clinica_file_reader(
             raise ClinicaBIDSError(error_message)
         else:
             raise ClinicaCAPSError(error_message)
-    return results
+    if len(error_encountered) > 0 and raise_exception is False:
+        error_message = (
+            f"Clinica encountered {len(error_encountered)} "
+            f"problem(s) while getting {information['description']}:\n"
+        )
+        if "needed_pipeline" in information.keys():
+            if information["needed_pipeline"]:
+                error_message += (
+                    "Please note that the following clinica pipeline(s) must "
+                    f"have run to obtain these files: {information['needed_pipeline']}\n"
+                )
+        for msg in error_encountered:
+            error_message += msg
+    return results, error_message
```

Comment:
There is a large potential for refactoring with the `raise_exception` case. Basically the error message construction code is the same, only what is done with it (raise in an exception or return the error message as a result) actually changes.

Reply:
That would make the code clearer : one section building the error message, one section returning or raising based on the input arguments.